### ISSUE TITLE
Issue with hover tool and invert_yaxis=True

### ISSUE THREAD

**Issue by Thalos12** [pyviz/holoviews#3834]
I am displaying an image in the form `Image((X,Y,Z))` as explained [here](http://holoviews.org/Reference_Manual/holoviews.element.html#holoviews.element.Image) with inverted y axis.
I also am using the hover tool to display raw values.

However, the region where the hover tool is actually working is not on the image, but below:

![holoviews_bug_cut](https://user-images.githubusercontent.com/13919407/61441009-0dbf7700-a945-11e9-86ab-6f80169da852.png)

The code I used to make this figure is

```python
%%opts Image [width=400, aspect=0.5, invert_yaxis=True] 
img = hv.Image( (cpos, mpos, sim.transpose()) ).opts(cmap=colorcet.fire)
img..opts(tools=['hover'], invert_yaxis=True)
```

I am using JupyterLab 1.0.2, Holoviews 1.12.3 and Bokeh 1.2.0

**Comment by poplarShift** [comment]
MWE if anybody else wants to try it: 
```
import holoviews as hv
import numpy as np
hv.extension('bokeh')

img = hv.Image((np.arange(10), np.arange(10), np.random.rand(10,10)))
img.opts(tools=['hover'], invert_yaxis=True, ylim=(0, 20))
```

**Comment by Thalos12** [comment]
@poplarShift Thanks for warning me about `%%opts`! 

I will only use `.opts` from now on, however I think `%%opts` was useful in case I had many similar figures to show in a Layout and did not want to manually write the options for each of them.
Probably there is a way that does not require `%%opts`? If there is, please excuse me, I started using `Holoviews` only recently.

**Comment by jbednar** [comment]
@Thalos12, you can use .opts with options you declare separately by type (as for %%opts), where they will later be applied only to the items of the declared type:

```
import holoviews as hv
import numpy as np
hv.extension('bokeh')

img = hv.Image((np.arange(10), np.arange(10), np.random.rand(10,10)))
opts = hv.opts.Image(tools=['hover'], invert_yaxis=True, ylim=(0, 20))
img.opts(lopts)
```

BTW, I"m curious where you ran across %%opts; we've tried to eliminate that from our examples to avoid confusion.

**Comment by Thalos12** [comment]
@jbednar Thanks for the suggestion, I will a variable to store the options then.

Regarding the `%%opts`, I likely copied some code from the internet and never discovered that it was deprecated!

**Comment by Thalos12** [comment]
@jbednar Actually, using the search bar of the website I found two places:

- in the [FAQ](http://holoviews.org/FAQ.html), towards the end (`Q: Why isn’t my %%opts cell magic being applied to my HoloViews object?`)
- in the [Notebook Magics](http://holoviews.org/user_guide/Notebook_Magics.html) section of the User guide.

I don't think I ever visited those pages before though.

**Comment by philippjfr** [comment]
Fixed in #4010

**Comment by Thalos12** [comment]
@philippjfr Thanks a lot!
